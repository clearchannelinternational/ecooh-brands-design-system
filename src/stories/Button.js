import React from 'react';
import PropTypes from 'prop-types';

/**
 * Primary UI component for user interaction
 */
export const Button = ({ mode, children, onClick, classes}) => {
  let classesToRender
  switch (mode) {
    case "primary":
      classesToRender = 'button-classic'
      break
    case "secondary":
      classesToRender = 'button-secondary'
      break
    default:
      classesToRender = ''
  }
  classesToRender += ` ${classes}`
  return (
    <button
      type="button"
      className={classesToRender}
      onClick={onClick}
    >
      {children || "Button"}
    </button>
  );
};

Button.propTypes = {
  /**
   * Additional css custom classes
   */
  classes: PropTypes.string,
  /**
   * Is this the principal call to action on the page?
   */
  mode: PropTypes.oneOf(['primary', 'secondary']),
  /**
   * Button contents
   */
  children: PropTypes.node.isRequired,
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
};

Button.defaultProps = {
  classes: '',
  mode: 'primary',
  onClick: ()=>true,
};
