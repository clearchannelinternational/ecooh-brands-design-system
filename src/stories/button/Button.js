import React from 'react';
import PropTypes from 'prop-types';

/**
 * Each button variant has a particular function and its design signals that function to the user. It is therefore very important that the different variants are implemented consistently across products, so that they message the correct actions.
*/
export const Button = ({ state, children, onClick, classes}) => {
  let classesToRender
  switch (state) {
    case "primary":
      classesToRender = 'es--btn es--btn--primary'
      break
    case "secondary":
      classesToRender = 'es--btn es--btn--secondary'
      break
    case "tertiary":
      classesToRender = 'es--btn es--btn--tertiary'
      break
    case "danger":
      classesToRender = 'es--btn es--btn--danger'
      break
    default:
      classesToRender = ''
  }
  classesToRender += ` ${classes}`
  return (
    <button
      type="button"
      className={classesToRender}
      onClick={onClick}
    >
      {children || "Button"}
    </button>
  );
};

Button.propTypes = {
  /**
   * Additional css custom classes
   */
  classes: PropTypes.string,
  /**
   * Is this the principal call to action on the page?
   */
  state: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'danger']),
  /**
   * Button contents
   */
  children: PropTypes.node.isRequired,
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
};

Button.defaultProps = {
  classes: '',
  state: 'primary',
  onClick: ()=>true,
};
