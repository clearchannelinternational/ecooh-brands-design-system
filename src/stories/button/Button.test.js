import React from 'react'
import { render, waitFor, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {Button} from './Button'

describe('As a user I want to have a button', () => {
  it('should render a primary button', () => {
    const { container } = render(
      <Button mode="primary" />
    )
    expect(container).toMatchSnapshot()
  })
  it('should render a secondary button', () => {
    const { container } = render(
      <Button mode="secondary" />
    )
    expect(container).toMatchSnapshot()
  })
  it('should allow additional classes', () => {
    const { container } = render(
      <Button mode="secondary" classes="custom custom2"/>
    )
    expect(container).toMatchSnapshot()
  })
  it('should allow custom content', () => {
    const { container } = render(
      <Button mode="secondary" >button test</Button>
    )
    expect(container).toMatchSnapshot()
  })
  it('should handle clicks', () => {
    const mockClick = jest.fn()
    render(
      <Button mode="secondary" onClick={mockClick}>button test</Button>
    )
    userEvent.click(screen.getByRole("button"))
    expect(mockClick).toHaveBeenCalled()
  })
})
