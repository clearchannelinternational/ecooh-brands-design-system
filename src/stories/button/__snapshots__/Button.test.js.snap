// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As a user I want to have a button should allow additional classes 1`] = `
<div>
  <button
    class="es--btn es--btn--secondary custom custom2"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`As a user I want to have a button should allow custom content 1`] = `
<div>
  <button
    class="es--btn es--btn--secondary "
    type="button"
  >
    button test
  </button>
</div>
`;

exports[`As a user I want to have a button should render a primary button 1`] = `
<div>
  <button
    class="es--btn es--btn--primary "
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`As a user I want to have a button should render a secondary button 1`] = `
<div>
  <button
    class="es--btn es--btn--secondary "
    type="button"
  >
    Button
  </button>
</div>
`;
